Sources/StreamChat/WebSocketClient/Events/MessageEvents.swift

```diff
@@ -63,13 +63,15 @@ public struct MessageReadEvent: EventWithUserPayload, EventWithChannelId {
     public let userId: UserId
     public let cid: ChannelId
     public let readAt: Date
+    public let unreadCount: UnreadCount?
     
     let payload: Any
     
     init<ExtraData: ExtraDataTypes>(from response: EventPayload<ExtraData>) throws {
         userId = try response.value(at: \.user?.id)
         cid = try response.value(at: \.cid)
         readAt = try response.value(at: \.createdAt)
+        unreadCount = try? response.value(at: \.unreadCount)
```

Comment:
Shouldn't we handle this field in `ChannelReadUpdaterMiddleware.resetChannelRead`?

Reply:
@b-onc  Not needed, this is already handled in `saveEvent` called by `EventDataProcessorMiddleware`

Reply:
`ChannelReadUpdaterMiddleware` handles channel unread counts, this is global unread count